### PR TITLE
Added service profile support in sc driver.

### PR DESCRIPTION
**Pull request by ashutosh-mishra**

### CODE REVIEW

**Comment by ashutosh-mishra** [comment]
@chandra-k @kedarkulkarni @Vikash082 @njagadish

File: gbpservice/neutron/services/servicechain/plugins/ncp/node_drivers/oneconvergence_service_node_driver.py [hunk 1]
```diff
@@ -1062,6 +1072,10 @@ def _fetch_template_and_params(self, context, update=False):
                                   'param': stack_params})
         return (stack_template, stack_params)
 
+    def _get_service_profile_type(self, context):
+        #return context.current_node['description']  # description should contain ASAV or VYOS
+        return SERVICE_PROFILE_TYPES[0]     # For now return 'ASAV'
```
Comment (rukansari) [inline gbpservice/neutron/services/servicechain/plugins/ncp/node_drivers/oneconvergence_service_node_driver.py]: We conveyed to sungard that service vendor or service flavor field will have the vendor type and HA. Not the  description field.
Why are we returning ASAV for now?

File: gbpservice/neutron/services/servicechain/plugins/ncp/node_drivers/oneconvergence_service_node_driver.py [hunk 2]
```diff
@@ -73,6 +73,7 @@
 
 STACK_ACTION_WAIT_TIME = cfg.CONF.oneconvergence_node_driver.stack_action_wait_time
 STACK_ACTION_RETRY_WAIT = 5  # Retry after every 5 seconds
+SERVICE_PROFILE_TYPES = ['ASAV', 'ASAV-HA', 'VYOS', 'VYOS-HA']
```
Comment (rukansari) [inline gbpservice/neutron/services/servicechain/plugins/ncp/node_drivers/oneconvergence_service_node_driver.py]: Why is haproxy and haproxy HA also not a type?

**Comment by rukansari** [comment]
Code not aligned to design approach conveyed to Sungard- vendor or flavor field to be used for vendor/HA. Not description.

**Comment by rukansari** [comment]
Please refer to Phase 2 design doc: sections -  Service Profile,  HA Support, workflow

File: gbpservice/neutron/services/servicechain/plugins/ncp/node_drivers/oneconvergence_service_node_driver.py (hunk 1 above)
Comment (ashutosh-mishra) [inline gbpservice/neutron/services/servicechain/plugins/ncp/node_drivers/oneconvergence_service_node_driver.py]: We are still not sure, how to get service_profile. We discussed here, that for now we'll return ASAV as service profile, until we don't have a way to chose between vyos and asav.
Commented code using description field is only for reference.

**Pull request closed without merging**